src/apply/patch/__test__/applyNodePatch.test.ts

```diff
@@ -30,6 +30,21 @@ describe('set', () => {
 
     assertType<{_key: string; title: string}[]>(result.objects)
   })
+
+  test('set same value inside object array', () => {
+    const document = {
+      objects: [
+        {_key: 'first', title: 'first'},
+        {_key: 'second', title: 'second'},
+      ],
+    }
+
+    const patch = at('objects[_key=="first"].title', set('first'))
+
+    const result = applyNodePatch(patch, document)
+
```

Comment:
Would be great to assert that the outcome of the patch is correct here too:
```suggestion
    expect(result).toEqual({
      objects: [
        {_key: 'first', title: 'first'},
        {_key: 'second', title: 'second'},
      ],
    })
```